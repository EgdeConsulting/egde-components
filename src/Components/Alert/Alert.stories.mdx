import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { DocsPageHeader, DocsPageBody } from 'Components';
import { LibPath } from 'Types';
import { Alert } from 'Components';

export const Template = (args) => <Alert {...args} />;

export const componentName = 'Alert';

<Meta
    title={LibPath.Feedback + componentName}
    component={Alert}
    args={{
        description: 'This is an info alert',
        status: 'info',
        linkText: '',
        linkHref: '',
        onClose: () => {},
        hasNoCloseButton: false,
        route: '',
    }}
/>

<DocsPageHeader
    componentName={componentName}
    componentDescription="This component is used to display a info-box which displays important information whom the user must see."
    addImportantNote={false}
    variant="stable"
/>

<Canvas withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Canvas>

<DocsPageBody addStories={true} />

### Warning Alert

<Canvas withToolbar>
    <Story
        name="Warning Alert"
        args={{
            description: 'This is a warning alert',
            status: 'warning',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Error Alert

<Canvas withToolbar>
    <Story
        name="Error Alert"
        args={{
            description: 'This is an error alert',
            status: 'error',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Success Alert

<Canvas withToolbar>
    <Story
        name="Success Alert"
        args={{
            description: 'This is a Success alert',
            status: 'success',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Alert with link

<Canvas withToolbar>
    <Story
        name="Alert with link"
        args={{
            description: 'This is an info alert',
            status: 'info',
            linkText: 'Learn more',
            linkHref: 'https://www.google.com',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
