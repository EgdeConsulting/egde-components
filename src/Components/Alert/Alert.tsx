import React, { ReactElement } from 'react';
import { AlertProps, IconTestID, AlertStatus } from 'Types';
import {
    Alert as ChakraAlert,
    AlertIcon,
    AlertDescription,
    CloseButton,
    Spacer,
    Stack,
    Link,
} from '@chakra-ui/react';
import {
    ErrorRounded,
    ErrorRoundedWarning,
    InfoRounded,
    CheckCircleRounded,
} from '../../Icons';


export const Alert = (props: AlertProps): ReactElement => {
    const { status, description, onClose, linkText, linkHref } = props;
    // styling should ideally be abstracted to ./Styles,
    // but didn't manage that with useMultiStyleConfig
    const styleComposition = {
        textColor: 'darkgrey',
        borderRadius: '6px',
        width: '400px',
        ...(status === AlertStatus.Warning && {
            backgroundColor: 'yellow',
            iconColor: 'amber',
            icon: ErrorRoundedWarning,
            iconTestID: IconTestID.AlertWarning,
        }),
        ...(status === AlertStatus.Info && {
            backgroundColor: 'lightblue',
            iconColor: 'blue',
            icon: InfoRounded,
            iconTestID: IconTestID.AlertInfo,
        }),
        ...(status === AlertStatus.Error && {
            backgroundColor: 'lightred',
            iconColor: 'red',
            icon: ErrorRounded,
            iconTestID: IconTestID.AlertError,
        }),
        ...(status === AlertStatus.Success && {
            backgroundColor: 'lightgreen',
            iconColor: 'green',
            icon: CheckCircleRounded,
            iconTestID: IconTestID.AlertSuccess,
        }),
    };

    return (
        <ChakraAlert
            status={status}
            color={styleComposition.textColor}
            colorScheme={styleComposition.backgroundColor}
            borderRadius={styleComposition.borderRadius}
            width={styleComposition.width}
        >
            <AlertIcon
                color={styleComposition.iconColor}
                as={styleComposition.icon}
                data-testid={styleComposition.iconTestID}
                position='absolute'
                left='12px'
                top='12px'
            />
            <Stack px='28px'>
                <AlertDescription>{description}</AlertDescription>
                {linkText && linkHref && (
                    <Link
                        href={linkHref}
                        color='blue'
                        textDecoration='underline'
                    >
                        {linkText}
                    </Link>
                )}
            </Stack>
            <Spacer />
            <CloseButton
                position='absolute'
                right='8px'
                top='8px'
                onClick={onClose}
            />
        </ChakraAlert>
    );
};
