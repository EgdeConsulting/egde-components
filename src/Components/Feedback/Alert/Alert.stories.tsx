/* import { Alert } from './Alert';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import { LibPath } from 'Types';
import { AlertStatus } from 'Styles/Components/SharedComponents';

export default {
  title: LibPath.Overlay + 'Alert' || 'undefined',
  component: Alert,
  parameters: {
    layout: 'centered',
  },
  argTypes: {
    status: { defaultValue: AlertStatus.Success },
    description: { defaultValue: 'This is a description' },
    onClose: { action: 'onclose' },
    linkText: { defaultValue: '' },
    linkHref: { defaultValue: '' },
  },
  args: {
    status: AlertStatus.Success,
    description: 'This is a description',
  },
} as ComponentMeta<typeof Alert>;

const Template: ComponentStory<typeof Alert> = (args) => <Alert {...args} />;

export const InfoAlert = Template.bind({});
InfoAlert.args = {
  description: 'This is an info alert',
  status: AlertStatus.Info,
};

export const WarningAlert = Template.bind({});
WarningAlert.args = {
  description: 'This is a warning alert',
  status: AlertStatus.Warning,
};

export const ErrorAlert = Template.bind({});
ErrorAlert.args = {
  description: 'This is an error alert',
  status: AlertStatus.Error,
};

export const SuccessAlert = Template.bind({});
SuccessAlert.args = {
  description: 'This is a success alert',
  status: AlertStatus.Success,
};

export const AlertWithLink = Template.bind({});
AlertWithLink.args = {
  description: 'This is a Error alert',
  status: AlertStatus.Error,
  linkText: 'Learn More',
  linkHref: 'https://www.google.com',
};
 */
// eslint-disable-next-line storybook/default-exports
