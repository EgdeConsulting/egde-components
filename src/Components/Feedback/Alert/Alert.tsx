import React, { ReactElement } from 'react';
import { AlertProps, IconTestID } from 'SharedTypes';
import {
    Alert as ChakraAlert,
    AlertIcon,
    AlertDescription,
    CloseButton,
    Spacer,
    Stack,
    Link,
    Text,
} from '@chakra-ui/react';
import { Link as ReactRouterLink } from 'react-router-dom';

export const Alert = (props: AlertProps): ReactElement => {
    const {
        status,
        description,
        onClose,
        linkText,
        linkHref,
        hasNoCloseButton,
        route,
    } = props;
    // styling should ideally be abstracted to ./Styles,
    // but didn't manage that with useMultiStyleConfig
    const styleComposition = {
        textColor: 'darkgrey',
        borderRadius: '0.375rem',
        width: '25.0rem',
        ...(status === 'warning' && {
            backgroundColor: 'yellow',
            iconColor: 'amber',
            iconTestID: IconTestID.AlertWarning,
        }),
        ...(status === 'info' && {
            backgroundColor: 'lightblue',
            iconColor: 'blue',
            iconTestID: IconTestID.AlertInfo,
        }),
        ...(status === 'error' && {
            backgroundColor: 'lightred',
            iconColor: 'red',
            iconTestID: IconTestID.AlertError,
        }),
        ...(status === 'success' && {
            backgroundColor: 'lightgreen',
            iconColor: 'green',
            iconTestID: IconTestID.AlertSuccess,
        }),
    };
    return (
        <ChakraAlert
            status={status}
            color={styleComposition.textColor}
            colorScheme={styleComposition.backgroundColor}
            borderRadius={styleComposition.borderRadius}
            width={styleComposition.width}
        >
            <AlertIcon
                color={styleComposition.iconColor}
                data-testid={styleComposition.iconTestID}
                position='absolute'
                left='0.75rem'
                top='0.75rem'
            />
            <Stack px='1.75rem'>
                <AlertDescription>{description}</AlertDescription>
                {linkText && linkHref && (
                    <Link
                        href={linkHref}
                        color='blue'
                        textDecoration='underline'
                    >
                        {linkText}
                    </Link>
                )}
                {route && linkText && (
                    <ReactRouterLink to={{ pathname: route }}>
                        <Text variant='link'>{linkText}</Text>
                    </ReactRouterLink>
                )}
            </Stack>
            <Spacer />
            {!hasNoCloseButton && (
                <CloseButton
                    position='absolute'
                    right='0.5rem'
                    top='0.5rem'
                    onClick={onClose}
                />
            )}
        </ChakraAlert>
    );
};
