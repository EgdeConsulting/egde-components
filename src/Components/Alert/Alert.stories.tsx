import { Alert } from "./Alert";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { LibPath } from "Types";


export default {
    title: LibPath.Overlay + 'Alert' || 'undefined',
    component: Alert,
    parameters: {
      layout: 'centered',
    },
    args: {
      status: 'info',
    },

} as ComponentMeta<typeof Alert>;


  const Template: ComponentStory<typeof Alert> = (args) => (
    <Alert {...args} />
  );

  export const InfoAlert = Template.bind({});
  InfoAlert.args = {
    description: 'This is an info alert',
  };
