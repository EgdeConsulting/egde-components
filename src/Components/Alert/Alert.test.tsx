import React from 'react';
import { render } from '@testing-library/react';
import { Alert } from './Alert';


// Snapshot Test
describe('Snapshot Alert', () => {
  it('should render correctly', () => {
    const tree = render(<Alert 
        description= 'This is an info alert' 
        status= 'info'
        linkText= ''
        linkHref= ''
        onClose={ () => {} }
        hasNoCloseButton ={false}
        route= ''
        />
        );
    expect(tree).toMatchSnapshot();
  });
});
