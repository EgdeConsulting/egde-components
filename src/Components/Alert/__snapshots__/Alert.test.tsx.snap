// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Alert should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-alert css-1pjqawk"
        role="alert"
      >
        <span
          class="chakra-alert__icon css-1h71qe"
          data-testid="alertInfoIcon"
        >
          <svg
            class="chakra-icon css-fi0ww0"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="chakra-stack css-1f2t4ws"
        >
          <div
            class="chakra-alert__desc css-6n7j50"
          >
            This is an info alert
          </div>
        </div>
        <div
          class="css-17xejub"
        />
        <button
          aria-label="Close"
          class="css-awkc72"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-alert css-1pjqawk"
      role="alert"
    >
      <span
        class="chakra-alert__icon css-1h71qe"
        data-testid="alertInfoIcon"
      >
        <svg
          class="chakra-icon css-fi0ww0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <div
        class="chakra-stack css-1f2t4ws"
      >
        <div
          class="chakra-alert__desc css-6n7j50"
        >
          This is an info alert
        </div>
      </div>
      <div
        class="css-17xejub"
      />
      <button
        aria-label="Close"
        class="css-awkc72"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
